src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -143,6 +144,11 @@ export class TerminalInstance implements ITerminalInstance {
 		if (platform.isWindows) {
 			this._processReady.then(() => {
 				this._windowsShellHelper = new WindowsShellHelper(this._processId, this._shellLaunchConfig.executable);
+				this._onCheckWindowsShell = new Emitter<TPromise<string>>();
+				debounceEvent(this._onCheckWindowsShell.event, (l, e) => e, 100, true)(() => {
+					this.checkWindowShell();
+				});
+				this.onData(() => this._onCheckWindowsShell.fire());
```

Comment:
I think we need a new mechanism for onData. Currently if a command runs for 10 seconds, it will launch 10/0.1=100 processes during its lifetimes. Instead of debounce like on enter we need something that will go off when the output has finished coming in.